File: src/maasserver/static/js/angular/factories/tests/test_resourcepools.js
/* Copyright 2018 Canonical Ltd.  This software is licensed under the
 * GNU Affero General Public License version 3 (see the file LICENSE).
 *
 * Unit tests for ResourcePoolsManager.
 */

import { makeInteger, makeName } from "testing/utils";

describe("ResourcePoolsManager", function() {
  // Load the MAAS module.
  beforeEach(angular.mock.module("MAAS"));

  // Load the ResourcePoolsManager.
  var ResourcePoolsManager;
  beforeEach(inject(function($injector) {
    ResourcePoolsManager = $injector.get("ResourcePoolsManager");
  }));

  function makeResourcePool(id) {
    var pool = {
      name: makeName("name")
    };
    if (angular.isDefined(id)) {
      pool.id = id;
    } else {
      pool.id = makeInteger(1, 100);
    }
    return pool;
  }

  it("set requires attributes", function() {
    expect(ResourcePoolsManager._pk).toBe("id");
    expect(ResourcePoolsManager._handler).toBe("resourcepool");
  });

  it("getDefaultPool returns pool with id = 0", function() {
    var defaultPool = makeResourcePool(0);
    ResourcePoolsManager._items.push(makeResourcePool());
    ResourcePoolsManager._items.push(defaultPool);
    expect(ResourcePoolsManager.getDefaultPool()).toBe(defaultPool);
  });
});

File: src/tests/maasservicelayer/models/test_resource_pools.py
#  Copyright 2025 Canonical Ltd.  This software is licensed under the
#  GNU Affero General Public License version 3 (see the file LICENSE).

import pytest

from maasservicelayer.models.resource_pools import ResourcePool


class TestResourcePool:
    @pytest.mark.parametrize(
        "id, is_default", [(-1, False), (0, True), (1, False)]
    )
    def test_is_default(self, id: int, is_default: bool):
        r = ResourcePool(id=id, name="", description="")
        assert r.is_default() is is_default
